test/e2e/inspection_test.go

```diff
@@ -0,0 +1,155 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/cluster-api/util"
+
+	bmov1 "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+)
+
+var _ = Describe("Inspection", func() {
+	var (
+		specName       = "inspection"
+		namespace      *corev1.Namespace
+		cancelWatches  context.CancelFunc
+		bmcUser        string
+		bmcPassword    string
+		bmcAddress     string
+		bootMacAddress string
+	)
+	BeforeEach(func() {
+		bmcUser = e2eConfig.GetVariable("BMC_USER")
+		bmcPassword = e2eConfig.GetVariable("BMC_PASSWORD")
+		bmcAddress = e2eConfig.GetVariable("BMC_ADDRESS")
+		bootMacAddress = e2eConfig.GetVariable("BOOT_MAC_ADDRESS")
+
+		namespace, cancelWatches = framework.CreateNamespaceAndWatchEvents(ctx, framework.CreateNamespaceAndWatchEventsInput{
+			Creator:   clusterProxy.GetClient(),
+			ClientSet: clusterProxy.GetClientSet(),
+			Name:      fmt.Sprintf("%s-%s", specName, util.RandomString(6)),
+			LogFolder: artifactFolder,
+		})
+	})
+
+	It("should put BMH without BMC credentials in unmanaged state", func() {
+		By("creating a BMH")
+		bmh := bmov1.BareMetalHost{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      specName + "-unmanaged",
+				Namespace: namespace.Name,
+			},
+		}
+		err := clusterProxy.GetClient().Create(ctx, &bmh)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("putting the BMH in unmanaged state")
```

Comment:
```suggestion
		By("waiting for the BMH to be in unmanaged state")
```

Reply:
Good point. I will update all of them in similar way